modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/property/ValuesProviderFactoryModelProperty.java

```diff
@@ -0,0 +1,217 @@
+   *                             to resolve the {@link Value values}
+   * @param connectionField      the field inside the {@link ValuesProvider} which is considered as a connection
+   * @param configField          the field inside the {@link ValuesProvider} which is considered as a configuration
+   */
+  private ValuesProviderFactoryModelProperty(Class<? extends ValuesProvider> valuesProvider,
+                                             List<InjectableParameterInfo> injectableParameters,
+                                             Field connectionField,
+                                             Field configField) {
+    checkNotNull(valuesProvider, "Values Provider Class parameter can't be null");
+    checkNotNull(injectableParameters, "injectableParameters parameter can't be null");
+
+    this.valuesProvider = valuesProvider;
+    this.injectableParameters = injectableParameters;
+    this.connectionField = connectionField;
+    this.configField = configField;
+  }
+
+  /**
+   * Creates a new builder to be able to easily build a {@link ValuesProviderFactoryModelProperty}
+   * @param valuesProvider the {@link Class} of a {@link ValuesProvider} implementation
+   *
+   * @return a new {@link ValuesProviderFactoryModelPropertyBuilder}
+   */
+  public static ValuesProviderFactoryModelPropertyBuilder builder(Class<? extends ValuesProvider> valuesProvider) {
+    return new ValuesProviderFactoryModelPropertyBuilder(valuesProvider);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String getName() {
+    return "ValuesProviderFactory";
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public boolean isPublic() {
+    return false;
+  }
+
+  /**
+   * @return the class of the {@link ValuesProvider} implementation
+   */
+  public Class<? extends ValuesProvider> getValueProvider() {
+    return valuesProvider;
+  }
+
+  /**
+   * @return The list of parameter names that are considered as required.
+   */
+  public List<String> getRequiredParameters() {
+    return injectableParameters
+        .stream()
+        .map(InjectableParameterInfo::getParameterName)
+        .collect(toList());
+  }
+
+  /**
+   * @return The {@link List} of parameters that requires to be inyected into the {@link ValuesProvider}
+   */
+  public List<InjectableParameterInfo> getInjectableParameters() {
+    return injectableParameters;
+  }
+
+  /**
+   * @return Indicates if the {@link ValuesProvider} requires a connection to resolve the {@link Value values}
+   */
+  public boolean usesConnection() {
+    return connectionField != null;
+  }
+
+  /**
+   * @return Indicates if the {@link ValuesProvider} requires a configuration to resolve the {@link Value values}
+   */
+  public boolean usesConfig() {
+    return configField != null;
+  }
+
+  /**
+   * @return The field inside the {@link ValuesProvider} that represents the connection
+   */
+  public Field getConnectionField() {
```

Comment:
why is this and the getter above needed? All this model property should give you is a Factory. All details should remain encapsulated